File: ReactASP.Domain/Entities/Language.cs
using System;
using System.Collections.Generic;

namespace ReactASP.Domain.Entities;

public partial class Language
{
    public Guid LangId { get; set; }

    public string Name { get; set; } = null!;

    public string Version { get; set; } = null!;

    public string RuntimeImage { get; set; } = null!;

    public string? CompileCommand { get; set; }

    public string? RunCommand { get; set; }

    public bool IsEnabled { get; set; }

    public virtual ICollection<Submission> Submissions { get; set; } = new List<Submission>();
}

File: meetcode.client/src/types/admin/language.ts
import { ProblemTemplate } from "./problemTemplate";
import { Submission } from "./submission";

export interface Language {
    langId: string;
    name: string;
    version: string;
    fileExtension: string;
    compileImage: string;
    runtimeImage: string;
    compileCommand: string | null;
    runCommand: string | null;
    isEnabled: boolean;
    submissions: Array<Submission>;
    problemTemplates: Array<ProblemTemplate>;
}